bond/bond.py

```diff
@@ -618,10 +618,14 @@ def _get_param_groups(files, layout, fieldmap_lookup, key_group_name):
         {"Counts": value_counts.to_numpy(),
          "ParamGroup": value_counts.index.to_numpy()})
 
-    param_groups_with_counts = pd.merge(
+    param_w_counts = pd.merge(
```

Comment:
the actual param group number assignment happens above on line 612. You'll need to actually reassign param group numbers __after__ you sort by the count